### ISSUE TITLE
Deriving SquircleBasedShape from CornerBasedShape

### ISSUE THREAD

**Issue by armond-avanes** [stoyan-vuchev/squircle-shape#18]
At the moment, because `SquircleBasedShape` is not a `CornerBasedShape`, you can't define a `androidx.compose.material3.Shapes` object consisting of `SquircleBasedShape` to pass into your `MaterialTheme` (to customize your component shapes at theme level). The reason is that `Shapes` class accepts `CornerBasedShape` as its constructor parameters. It will be very helpful if you make this change....

```
abstract class SquircleBasedShape(
    topStart: CornerSize,
    topEnd: CornerSize,
    bottomStart: CornerSize,
    bottomEnd: CornerSize,
    val cornerSmoothing: Float
) : CornerBasedShape(
    topStart = topStart,
    topEnd = topEnd,
    bottomStart = bottomStart,
    bottomEnd = bottomEnd
)
```

And no need to create overloaded versions of `createOutline()` and `copy()` methods to pass in the extra `cornerSmoothing` parameter. Wherever needed in `SquircleShape` class, you can refer to the `val` field passed into `SquircleBasedShape` constructor above.

**Comment by stoyan-vuchev** [comment]
Thank you for bringing this up! Your suggestion to make `SquircleBasedShape` extend `CornerBasedShape` is very insightful and aligns well with how Compose themes are typically structured. This change would indeed allow `SquircleBasedShape` to integrate seamlessly into the `Shapes` class, enabling users to define squircle shapes at the theme level, which is currently not possible.

### Proposed Changes:
1. **Inheritance from `CornerBasedShape`**: 
   Making `SquircleBasedShape` extend `CornerBasedShape` would allow it to be used directly in `androidx.compose.material3.Shapes`. This would be a clean way to integrate squircle shapes into the Material Design ecosystem without additional workarounds.
   
2. **Usage of `cornerSmoothing` as a Parameter**: 
   Your approach to storing the `cornerSmoothing` parameter as a `val` field in the `SquircleBasedShape` constructor is practical and avoids the need for overloaded methods like `createOutline()` and `copy()`.

### Implementation Considerations:
While this approach simplifies the integration, I’ll need to carefully review any potential implications this might have, such as:
- **Backwards Compatibility**: Ensuring that existing usages of `SquircleBasedShape` are unaffected by this change.
- **Performance**: Verifying that adding the `cornerSmoothing` property does not introduce any unintended overhead in shape rendering.

### Next Steps:
I'll evaluate this suggestion further and test its integration with `MaterialTheme` and other use cases to confirm smooth functionality. If everything works as expected, I’ll prepare a PR to incorporate this enhancement.

Thanks again for pointing this out and for the detailed explanation! Your input is greatly appreciated and helps make this library more flexible and user-friendly.

**Comment by stoyan-vuchev** [comment]
Thank you for your suggestion and detailed explanation! 🎉

I’ve implemented the changes in version **3.0.0**, where `SquircleBasedShape` now extends `CornerBasedShape`, allowing it to be seamlessly integrated into `androidx.compose.material3.Shapes`. Additionally, the `cornerSmoothing` parameter is now directly part of the constructor, streamlining its usage without the need for overloaded methods.

You can now define squircle shapes at the theme level and customize your components more flexibly. Please update to version **3.0.0** and let me know if you encounter any issues or have further suggestions.

Thanks again for contributing to the improvement of the library! 🙌

**Comment by armond-avanes** [comment]
Thank you for accepting my suggestion and for your prompt action! It's working perfect now!
